### PR TITLE
Feature/46956 - Migrate typography

### PR DESCRIPTION
**Pull request by jszczech**

### CODE REVIEW

File: components/01-globals/blocquote/Blockquote.vue [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<template>
+  <blockquote>
+    {{ blockquote }}
+    <cite
```
Comment (Igloczek) [inline components/01-globals/blocquote/Blockquote.vue]: `cite` is not part of `blockquote`

File: components/01-globals/blocquote/Blockquote.vue [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<template>
+  <blockquote>
+    {{ blockquote }}
```
Comment (Igloczek) [inline components/01-globals/blocquote/Blockquote.vue]: Content should be passed via slot, we don't want to limit users in any way, nor use `v-html`

File: components/01-globals/blocquote/Blockquote.vue [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    <cite
+      v-if="cite">
+      <a
```
Comment (Igloczek) [inline components/01-globals/blocquote/Blockquote.vue]: Links are definitely not a part of the quote element

File: components/01-globals/grid/GridCollumn.vue [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <div :class="['grid__column', width ? `grid__column--${width}` : '']">
+  <div :class="['grid__column', width && `grid__column--${width}`]">
```
Comment (Igloczek) [inline components/01-globals/grid/GridCollumn.vue]: This is weird and probably doesn't work anymore, b/c `&&` return Boolean, while we need String.

File: components/01-globals/preformatted/Preformatted.vue [hunk 5]
```diff
@@ -0,0 +1,36 @@
+<template>
+  <component
+    :is="tag ? tag : 'pre'"
+    :class="tag && 'preformatted'">
```
Comment (Igloczek) [inline components/01-globals/preformatted/Preformatted.vue]: We need to tweak prettier to move closing `>` to the new line.

File: components/01-globals/preformatted/Preformatted.vue [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    <slot/>
+  </component>
+</template>
+
+<script>
+export default {
+  props: {
+    tag: {
+      type: String,
+      default: null
```
Comment (Igloczek) [inline components/01-globals/preformatted/Preformatted.vue]:
```suggestion
      default: 'pre'
```

File: components/01-globals/preformatted/Preformatted.vue [hunk 7]
```diff
@@ -0,0 +1,36 @@
+<template>
+  <component
+    :is="tag ? tag : 'pre'"
```
Comment (Igloczek) [inline components/01-globals/preformatted/Preformatted.vue]:
```suggestion
    :is="tag"
```

File: components/01-globals/typography/Heading.vue [hunk 8]
```diff
@@ -1,3 +1,12 @@
+<template>
+  <component
+    :is="tag ? tag : 'h' + level"
+    :class="[tag && `heading heading--${levelClassName}-level`,
+             page && 'heading--page']">
```
Comment (Igloczek) [inline components/01-globals/typography/Heading.vue]: Dear Prettier, this formatting is 💩 , I'm disappointed :<

File: components/01-globals/typography/Heading.vue [hunk 9]
```diff
@@ -1,3 +1,12 @@
+<template>
+  <component
+    :is="tag ? tag : 'h' + level"
+    :class="[tag && `heading heading--${levelClassName}-level`,
```
Comment (Igloczek) [inline components/01-globals/typography/Heading.vue]: Why do you check `tag` if you need `levelClassName`?

File: components/01-globals/typography/Heading.vue (hunk 8 above)
Comment (Igloczek) [reply components/01-globals/typography/Heading.vue]: IMO it should looks like this

```vue
<component
  :is="tag"
  :class="{
    [`heading heading--${levelClassName}-level`]: levelClassName,
    'heading--page': page
  }"
>
```

File: components/01-globals/typography/Label.vue [hunk 10]
```diff
@@ -1,19 +1,18 @@
+<template>
+  <component
+    :is="tag ? tag : 'label'"
+    :class="[tag && 'label', {
```
Comment (Igloczek) [inline components/01-globals/typography/Label.vue]: why `label` is done in array way, but others in the object, while are doing same thing :D?

File: components/02-elements/button/Button.vue [hunk 11]
```diff
@@ -1,34 +1,26 @@
+<template>
+  <component
+    :is="tag ? tag : 'button'"
+    :class="[tag && 'button',
+             customClass && customClass,
+             icon && 'button--icon',
+             secondary && 'button--secondary',
+             link && 'button--link',
+             fluid && 'button--fluid']">
```
Comment (Igloczek) [inline components/02-elements/button/Button.vue]: It should be in object, not array, will be way cleaner

Comment (Igloczek) [reply components/02-elements/button/Button.vue]: I'm not sure if that's that option (b/c we don't use JSX), but that's what we need to set to `false` https://prettier.io/docs/en/options.html#jsx-brackets

File: components/02-elements/button/Button.vue [hunk 12]
```diff
@@ -1,34 +1,26 @@
...
+    <slot/>
+    <icon
+      v-if="icon"
+      :icon="icon"
+      :class="'button__icon'"/>
```
Comment (Igloczek) [inline components/02-elements/button/Button.vue]:
```suggestion
      class="button__icon"
    />
```

File: components/02-elements/button/Button.vue [hunk 13]
```diff
@@ -1,34 +1,26 @@
+<template>
+  <component
+    :is="tag ? tag : 'button'"
+    :class="[tag && 'button',
```
Comment (Igloczek) [inline components/02-elements/button/Button.vue]: `button` class is not conditional

File: components/01-globals/typography/Paragraph.vue [hunk 14]
```diff
@@ -1,16 +1,17 @@
...
-        class: {
-          paragraph: this.tag
-        }
-      },
-      this.$slots.default
-    )
+  props: {
+    tag: {
+      type: String,
+      default: null
```
Comment (Igloczek) [inline components/01-globals/typography/Paragraph.vue]:
```suggestion
      default: 'p'
```

File: components/01-globals/typography/Heading.vue (hunk 9 above)
Comment (jszczech) [reply components/01-globals/typography/Heading.vue]: B/c when we have a tag, e.g. `h6,` in styles in tag we extend `heading` and `heading--${levelClassName}-level`. It does not make sense to add classes, when we extend this styles inside the tag. If we want to add this classes, I think, we should remove a part of styles. :)

File: components/01-globals/grid/GridCollumn.vue (hunk 4 above)
Comment (jszczech) [reply components/01-globals/grid/GridCollumn.vue]: It works fine this way :)

File: components/02-elements/button/Button.vue (hunk 13 above)
Comment (jszczech) [reply components/02-elements/button/Button.vue]: Hmm, why not? If `tag === 'button'`, we don't need add this class.

File: .eslintrc.js [hunk 15]
```diff
@@ -15,6 +15,10 @@ module.exports = {
   },
   rules: {
     "vue/require-default-prop": 1,
-    "vue/require-prop-types": 1
+    "vue/require-prop-types": 1,
+    "vue/html-closing-bracket-newline": [2, {
```
Comment (Igloczek) [inline .eslintrc.js]: Create an issue in prettier to handle this thing in a correct way

**Pull request merged**